finetune/target_models/seq2seq.py

```diff
@@ -0,0 +1,139 @@
+
+import tensorflow as tf
+
+from finetune.base import BaseModel
+from finetune.encoding.target_encoders import Seq2SeqLabelEncoder
+from finetune.input_pipeline import BasePipeline
+from finetune.util.shapes import shape_list
+from finetune.nn.target_blocks import language_model
+from finetune.model import PredictMode
+from finetune.util.beam_search import beam_search
+
+
+class S2SPipeline(BasePipeline):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.target_dim = 0
+        self.label_encoder = self._target_encoder()
+
+    def feed_shape_type_def(self):
+        TS = tf.TensorShape
+        return (
+            (
+                {
+                    "tokens": tf.int32,
+                },
+                (tf.int32, tf.int32)
+            ),
+            (
+                {
+                    "tokens": TS([None]),
+                },
+                (TS([None]), TS([]))
+            )
+        )
+
+    def _target_encoder(self):
+        return Seq2SeqLabelEncoder(self.text_encoder, self.config.max_length)
+
+def normalize_embeds(embeds):
+    return embeds * (tf.cast(tf.shape(embeds)[-1], dtype=tf.float32) ** -0.5)
+
+class HFS2S(BaseModel):
+
```

Comment:
Future work -- exposing option to provide inputs to decoder that are conditioned on (at train or test time) but act as a prefix / aren't part of the loss.